src/python/grpcio_observability/grpc_observability/_cyobservability.pyx

```diff
@@ -0,0 +1,249 @@
+# Copyright 2023 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cimport cpython
+from cython.operator cimport dereference
+
+import logging
+from threading import Thread
+from typing import List, Mapping, Tuple
+
+import grpc_observability
+
+
+cdef const char* CLIENT_CALL_TRACER = "client_call_tracer"
+cdef const char* SERVER_CALL_TRACER_FACTORY = "server_call_tracer_factory"
+cdef bint GLOBAL_SHUTDOWN_EXPORT_THREAD = False
+cdef object global_export_thread
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MetricsName:
+  CLIENT_API_LATENCY = kRpcClientApiLatencyMeasureName
+  CLIENT_SNET_MESSSAGES_PER_RPC = kRpcClientSentMessagesPerRpcMeasureName
+  CLIENT_SEND_BYTES_PER_RPC = kRpcClientSentBytesPerRpcMeasureName
+  CLIENT_RECEIVED_MESSAGES_PER_RPC = kRpcClientReceivedMessagesPerRpcMeasureName
+  CLIENT_RECEIVED_BYTES_PER_RPC = kRpcClientReceivedBytesPerRpcMeasureName
+  CLIENT_ROUNDTRIP_LATENCY = kRpcClientRoundtripLatencyMeasureName
+  CLIENT_SERVER_LATENCY = kRpcClientServerLatencyMeasureName
+  CLIENT_STARTED_RPCS = kRpcClientStartedRpcsMeasureName
+  CLIENT_RETRIES_PER_CALL = kRpcClientRetriesPerCallMeasureName
+  CLIENT_TRANSPARENT_RETRIES_PER_CALL = kRpcClientTransparentRetriesPerCallMeasureName
+  CLIENT_RETRY_DELAY_PER_CALL = kRpcClientRetryDelayPerCallMeasureName
+  CLIENT_TRANSPORT_LATENCY = kRpcClientTransportLatencyMeasureName
+  SERVER_SENT_MESSAGES_PER_RPC = kRpcServerSentMessagesPerRpcMeasureName
+  SERVER_SENT_BYTES_PER_RPC = kRpcServerSentBytesPerRpcMeasureName
+  SERVER_RECEIVED_MESSAGES_PER_RPC = kRpcServerReceivedMessagesPerRpcMeasureName
+  SERVER_RECEIVED_BYTES_PER_RPC = kRpcServerReceivedBytesPerRpcMeasureName
+  SERVER_SERVER_LATENCY = kRpcServerServerLatencyMeasureName
+  SERVER_STARTED_RPCS = kRpcServerStartedRpcsMeasureName
+
+# Delay map creation due to circular dependencies
+_CY_METRICS_NAME_TO_PY_METRICS_NAME_MAPPING = {}
+
+def cyobservability_init(object exporter) -> None:
+  global _CY_METRICS_NAME_TO_PY_METRICS_NAME_MAPPING
+  _CY_METRICS_NAME_TO_PY_METRICS_NAME_MAPPING = {x.value: x for x in grpc_observability.MetricsName}
```

Comment:
This could be done at module initialization time instead of during this function call by just moving it up a few lines. This would avoid the runtime modification of global mutable state (signalled by the `global` keyword), which should be done sparingly.

Reply:
That's what's I did at first, but we'll have an circular import issue: This map needs `grpc_observability.MetricsName` and `grpc_observability.MetricsName`  requires `Cython.MetricsName`.

I moved `grpc_observability.MetricsName` to Cython file so we don't have circular import anymore, but the downside is `MetricsName` which should be a public class is now imported from a Cython file.

Reply:
> but we'll have an circular import issue

That makes sense. Please add a comment here explaining why this is being done.

> but the downside is MetricsName which should be a public class is now imported from a Cython file

Besides this hack to get around the circular import, that doesn't seem like a big deal to me. Could you maybe instead move the definition of `grpc_observability.MetricsName` to this file (or some new third file) and re-import it in `__init__.py`?

Reply:
That's what I did now, both `MetricsName` and `CyMetricsName` are in `.pyx` file and `MetricsName` was imported to `__init__.py`.